### ISSUE TITLE
Toolbar in Card has some layout issues

### ISSUE THREAD

**Issue by theolof** [CosmicMind/Material#677]
When adding
`toolbar.interimSpacePreset = .interimSpace6`
to a Card, it messes up the toolbar. It pushes the leftViews and rightViews outside the toolbar boundaries. See screenshots.
<img width="625" alt="screen shot 2017-02-05 at 14 01 14" src="https://cloud.githubusercontent.com/assets/14074538/22626379/308e5a34-ebac-11e6-8ba7-20b856d982f7.png">
<img width="621" alt="screen shot 2017-02-05 at 14 02 12" src="https://cloud.githubusercontent.com/assets/14074538/22626378/308e24ce-ebac-11e6-841f-b90ff48edb83.png">

**Comment by danieldahan** [comment]
Hey :) 

The issue is that the calculation is based on the height, and since it is a circle, the width is adjusting to the size of the height. So it is pushing the image out. I will take a look at this now and see if I can add a fix.

**Comment by theolof** [comment]
Here's two more screenshots of the problem.

<img width="679" alt="screen shot 2017-02-09 at 12 01 46" src="https://cloud.githubusercontent.com/assets/14074538/22780633/a536a29a-eebf-11e6-93c3-0881ef1084fe.png">
Two problems for me in the above shot - First: I add a tap gesture to the red area (which should contain the knife/fork image), but since it's smaller than the image, it's hard to click. Second: The image is pushed outside the toolbar.


<img width="671" alt="screen shot 2017-02-09 at 11 42 07" src="https://cloud.githubusercontent.com/assets/14074538/22779883/d59006b4-eebc-11e6-936d-66436d4db250.png">

In short, I do the following for the shot above:
```
var responseView: UIView = UIView()
toolbar.rightViews = [responseView]
```
where responseView contains two FlatButtons

The responseView is red in the screenshot, it doesn't expand to the size of the buttons. Therefore (?) the buttons lives outside the toolbar. The toolbar is green.

Any suggestions/solutions? I'm pulling my hair about this, I really need to get this to work...

**Comment by theolof** [comment]
I can solve it somewhat by adding the buttons directly to the rightViews:
`toolbar.rightViews = [checkMarkButton, ignoreButton]`
but then some large spacing occurs, see below.

<img width="622" alt="screen shot 2017-02-09 at 12 35 44" src="https://cloud.githubusercontent.com/assets/14074538/22781875/ce93c12c-eec4-11e6-8bf0-1c59aaec67f2.png">

The same happens if I add a `View` directly to the rightViews/leftViews instead of wrapping the `View` in a `UIView` - it then sits nicely inside the toolbar, but the spacing gets messed up.

<img width="619" alt="screen shot 2017-02-09 at 12 45 56" src="https://cloud.githubusercontent.com/assets/14074538/22782093/cf815d50-eec5-11e6-88dc-ca7b6fd14a6a.png">

**Comment by danieldahan** [comment]
Working more on this today :)

**Comment by danieldahan** [comment]
Hey, I am working on your issue using the `issue-677` branch. So far coming along.

**Comment by danieldahan** [comment]
So I reviewed the issue and I realize there isn't an issue with Material. Your settings need to adjust. In the last try you set the values directly without wrapping them. That is correct. To move the title label over, you need to set the `toolbar.contentViewAlignment` property to `.left` or set the `titleLabel.textAlignment` value to `.left`. To create symmetry, the `contentView` holds the `titleLabel` and `detailLabel` and balances their spacing based on the opposite side views.

If you still have an issue, reopen this one. Thank you!

**Comment by theolof** [comment]
`toolbar.contentViewAlignment` doesn't have a property `.left`, only `.full` or `.center` and they don't help me.
I'm also not using `titleLabel` or `detailLabel` in this case.

Here's what I'm trying now:

profileImageView = View()
profileImageView.shapePreset = .circle
profileImageView.image = profileImage.resize(toWidth: PROFILE_IMAGE_WIDTH)!

activityTextLabel = TTTAttributedLabel(frame: CGRect.zero) // tried with a regular UILabel - no difference

toolbar = Toolbar()
toolbar.interimSpacePreset = .interimSpace6
toolbar.leftViews = [profileImageView]
toolbar.centerViews = [activityTextLabel]
toolbar.rightViews = [placeImageView]

**Comment by danieldahan** [comment]
The `centerViews` is in the `contentView` area. So I would still set it to `.full`. Can you share a quick sample so I can see what is up?

**Comment by theolof** [comment]
Didn't find the time today, but I think I can do it tomorrow. Cheers.

**Comment by theolof** [comment]
Here's a sample. I modified your PresenterCard sample.
[PresenterCard.tar.gz](https://github.com/CosmicMind/Material/files/782656/PresenterCard.tar.gz)

This is the result:
Also notice that the right image is skewed, it should be a globe. Don't know how to make it follow the aspect ratio of the original image.
![screen shot 2017-02-17 at 09 42 37](https://cloud.githubusercontent.com/assets/14074538/23058601/1389dc58-f4f6-11e6-83d6-f52d3f0638e0.png)
With "Debug->Color Blended Layers" on:
![screen shot 2017-02-17 at 09 45 30](https://cloud.githubusercontent.com/assets/14074538/23058619/2a728ed8-f4f6-11e6-9047-1971b3a24c34.png)

**Comment by danieldahan** [comment]
Cool, I will take a look. Thank you!

**Comment by danieldahan** [comment]
Hey, is this still an issue? If yes, please reopen it, otherwise sorry for any delays and inconvenience.

**Comment by theolof** [comment]
Yes, it's still behaving the same.

I'm not using the toolbar due to this (though I would like to, for convenience)

**Comment by danieldahan** [comment]
Okay, I am making an update to the `RootController` types, which includes the `ToolbarController`. I am going to reopen this issue and investigate further. Thank you!

**Comment by OrkhanAlikhanov** [comment]
@danieldahan I investigated this issue a bit with [PresenterCard.tar.gz](https://github.com/CosmicMind/Material/files/782656/PresenterCard.tar.gz) sample that @theolof had attached. The project was using version 2.4.16, I tested with it and latest one, both was kinda misbehaving when `View` is used. I couldn't find the cause of this. I am attaching a [minimal project](https://github.com/CosmicMind/Material/files/2195193/PresenterCard.zip) showing the so-called bug. Here is how it looks, red ones are `View` instances whereas blue ones are `UIImageView` instances.

![image](https://user-images.githubusercontent.com/15037839/42727613-3d21aa2e-87bb-11e8-85a6-fcd72a8a3f38.png)

**Comment by danieldahan** [comment]
If a UIViewImage is used, it looks correct. @theolof thoughts?
